File: dart/06-io-operations/io_format.dart
import 'dart:io';

void main() {
  double pi = 3.14159265;

  // Fixed number of decimal places
  print('Pi to 2 places: ${pi.toStringAsFixed(2)}');

  // Build an aligned table with padding
  List<List<String>> rows = [
    ['Apple', '3'],
    ['Banana', '12'],
    ['Cherry', '150'],
  ];

  for (var row in rows) {
    String name = row[0].padRight(10);  // left-align in 10 columns
    String qty = row[1].padLeft(5);     // right-align in 5 columns
    print('$name|$qty');
  }

  // Convert numbers to other bases and pad with zeros
  int value = 42;
  print('Hex: ${value.toRadixString(16)}');
  print('Binary: ${value.toRadixString(2)}');
  print('Padded: ${value.toString().padLeft(6, '0')}');
}

File: nim/06-io-operations/io_format.nim
import std/strformat

let name = "Nim"
let version = 2.2
let count = 7
let pi = 3.14159

# Interpolate variables directly into the string
echo fmt"Language: {name}, version {version}"

# Left-align a label to width 12, right-align a number to width 4
let label = "Widgets"
echo fmt"{label:<12}| {count:>4}"

# Floating point with two decimal places
echo fmt"Pi to 2 dp: {pi:.2f}"

# The same integer in hexadecimal and binary
echo fmt"{count} in hex = {count:x}, in binary = {count:b}"
